images/cellregmap/summarise.py

```diff
@@ -0,0 +1,184 @@
+import os
+import click
+import pandas as pd
+import numpy as np
+
+def smartAppend(table,name,value):
+    """ helper function for appending in a dictionary """
+    if name not in table.keys():
+        table[name] = []
+    table[name].append(value)
+
+def qv_estimate(pv, m=None, verbose=False, lowmem=False, pi0=None):
+    """
+    Estimates q-values from p-values
+    Args
+    =====
+    m: number of tests. If not specified m = pv.size
+    verbose: print verbose messages? (default False)
+    lowmem: use memory-efficient in-place algorithm
+    pi0: if None, it's estimated as suggested in Storey and Tibshirani, 2003.
+         For most GWAS this is not necessary, since pi0 is extremely likely to be
+         1
+
+    Taken from https://github.com/nfusi/qvalue
+    but updated for python 3 (e.g., xrange -> range)
+
+    Copyright (c) 2012, Nicolo Fusi, University of Sheffield
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+        * Redistributions of source code must retain the above copyright
+        notice, this list of conditions and the following disclaimer.
+        * Redistributions in binary form must reproduce the above copyright
+        notice, this list of conditions and the following disclaimer in the
+        documentation and/or other materials provided with the distribution.
+        * Neither the name of the organization nor the
+        names of its contributors may be used to endorse or promote products
+        derived from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+    ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+    WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER BE LIABLE FOR ANY
+    DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+    (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+    ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+    """
+
+    import scipy as sp
+    from scipy import interpolate
```

Comment:
These imports should be at the top of the file

Reply:
i wanted to keep track of the fact that these are part of this function that's why, but I see how your solution is cleaner